server/api_group.go

```diff
@@ -289,3 +294,35 @@ func (s *ApiServer) ListUserGroups(ctx context.Context, in *api.ListUserGroupsRe
 
 	return userGroups, nil
 }
+
+func (s *ApiServer) ListGroups(ctx context.Context, in *api.ListGroupsRequest) (*api.GroupList, error) {
+	limit := 1
+	if in.GetLimit() != nil {
+		if in.GetLimit().Value < 1 || in.GetLimit().Value > 100 {
+			return nil, status.Error(codes.InvalidArgument, "Invalid limit - limit must be between 1 and 100.")
+		}
+		limit = int(in.GetLimit().Value)
+	}
+
+	cursorStr := in.GetCursor()
+	var cursor *groupListCursor = nil
+	if cursorStr != "" {
```

Comment:
We should do cursor processing inside the core function, just like the other cases. It's mostly useful when we expose the listing op in the runtime.